gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/resource/integration/IntegrationResource.java

```diff
@@ -172,9 +176,36 @@ public IngestionPreviewResponse previewNewIntegrationApis(@PathParam("integratio
 
         AuditInfo audit = getAuditInfo();
 
-        var output = previewNewIntegrationApisUseCase.execute(new PreviewNewIntegrationApisUseCase.Input(integrationId, audit));
+        return discoveryUseCase
+            .execute(new DiscoveryUseCase.Input(integrationId, audit))
+            .map(IntegrationResource::mapper)
+            .toCompletionStage();
+    }
 
-        return IngestionPreviewResponse.builder().totalCount(output.newApisCount()).build();
+    private static IngestionPreviewResponse mapper(DiscoveryUseCase.Output preview) {
```

Comment:
I guess you can add a method [here](https://github.com/gravitee-io/gravitee-api-management/blob/master/gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/mapper/IntegrationMapper.java) to let MapStruct building the mapper